tests/FakeItEasy.Specs/ThenSpecs.cs

```diff
@@ -0,0 +1,163 @@
+                });
+
+            "Then the delegate is invoked twice"
+                .x(() => A.CallTo(() => action()).MustHaveHappened(Repeated.Exactly.Twice));
+
+            "And the third call throws an exception"
+                .x(() => exception.Should().BeAnExceptionOfType<InvalidOperationException>());
+        }
+
+        [Scenario]
+        public void NonVoidMethodWithReturns(
+            IFoo foo,
+            int result1,
+            int result2,
+            int result3)
+        {
+            "Given a fake"
+                .x(() => foo = A.Fake<IFoo>());
+
+            "When a non-void method is configured to return 1 once then return 2"
+                .x(() =>
+                    A.CallTo(() => foo.Baz()).Returns(1).Once()
+                        .Then.Returns(2));
+
+            "And the method is called 3 times"
+                .x(() =>
+                {
+                    result1 = foo.Baz();
+                    result2 = foo.Baz();
+                    result3 = foo.Baz();
+                });
+
+            "Then the first call returns 1"
+                .x(() => result1.Should().Be(1));
+
+            "And the second call returns 2"
+                .x(() => result2.Should().Be(2));
+
+            "And the third call returns 2"
+                .x(() => result3.Should().Be(2));
+        }
+
+        [Scenario]
+        public void AnyCallWithInvokesAndThrows(
+            IFoo foo,
+            Action action1,
+            Action action2,
+            Exception exception)
+        {
+            "Given a fake"
+                .x(() => foo = A.Fake<IFoo>());
+
+            "And two delegates"
+                .x(() =>
+                {
+                    action1 = A.Fake<Action>();
+                    action2 = A.Fake<Action>();
+                });
+
+            "When any method is configured to invoke a delegate once, then another delegate twice, then throw an exception"
+                .x(() =>
+                    A.CallTo(foo).Invokes(action1).DoesNothing().Once()
+                        .Then.Invokes(action2).DoesNothing().Twice()
+                        .Then.Throws<InvalidOperationException>());
+
+            "And 4 calls are made on the fake"
+                .x(() =>
+                {
+                    foo.Bar();
+                    foo.Bar(1);
+                    foo.Bar(2);
+                    exception = Record.Exception(() => foo.Bar(3));
+                });
+
+            "Then the first delegate is called once"
+                .x(() => A.CallTo(() => action1()).MustHaveHappened(Repeated.Exactly.Once));
+
+            "And the second delegate is called twice"
+                .x(() => A.CallTo(() => action2()).MustHaveHappened(Repeated.Exactly.Twice));
+
+            "And the fourth call throws an exception"
+                .x(() => exception.Should().BeAnExceptionOfType<InvalidOperationException>());
+        }
+    }
+}

```

Comment:
I wonder if there shouldn't be some specs to show the interaction between the `Then`-style specifications and the old-style "define behaviour, then defined overriding behaviour", just to make sure it keeps working.
We won't encourage this kind of thing but, it will come up via implicit rules and/or `IFakeOptionBuilder`s.